atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/module/Table.groovy

```diff
@@ -175,6 +178,39 @@ class Table {
         return new Range(service, service.getRange(query, token) as Map, token)
     }
 
+    /**
+     * Lazily joins against an Iterable of key/value pairs.
+     *
+     *
+     *
+     * Tip: Since it is hard to lazily transform Iterables in Groovy, use Guava.
+     *
+     * Example:
+     * <pre>
+     * FluentIterable = com.google.common.collect.FluentIterable
+     * input = FluentIterable.from(table("myTable").getRange()).transform{ [(it.row): it}
+     * output = table("myOtherTable").join(input)
+     * </pre>
+     * @param input Iterable of Map<joinKey, inputValue> where JOIN_KEY is the row key of this table.
+     * @param cols columns to select from this table.
+     * @param batchSize size of getRows calls against this table, default 1000.
+     * @param token.
+     * @return Iterable of Maps with size 3, structured as
+     * <pre>
+     * { JOIN_KEY: joinKey, INPUT_VALUE: inputValue, OUTPUT_VALUE: outputValue}
+     * </pre>
+     * where joinKey and inputValue map to the inputs, and outputValue is the corresponding row in this table.
+     */
+    public Iterable<Map<String, Object>> join(
+            Iterable<Map<?, ?>> input,
+            cols = null,
+            int batchSize = 10000,
+            TransactionToken token = service.getTransactionToken()) {
+
+        return AtlasConsoleJoins.join(input, batchSize, { keys -> getRows(keys, cols, token)});
```

Comment:
Unclear to me -- not documented/guaranteed anywhere I can see.  Also unclear if that would mean the order you pass in, or the natural ordering of the keys.  I didn't want to rely on that.